sherpa/astro/ui/serialize.py

```diff
@@ -721,26 +827,25 @@ def _handle_usermodel(mod, modelname, fh=None):
     # in case getsource can return None, have check here
     if pycode is None:
         msg = "Unable to save Python code for user model " + \
-              "'{}' function {}".format(modelname, mod.calc.name)
+              f"'{modelname}' function {mod.calc.__name__}"
         warning(msg)
-        _output('print("{}")'.format(msg), fh)
-        _output("def {}(*args):".format(mod.calc.name), fh)
-        _output("    raise NotImplementedError('User model was " +
-                "not saved by save_all().'", fh)
-        _output("", fh)
+        _output(out, f'print("{msg}")')
+        _output(out, f"def {mod.calc.__name__}(*args):")
+        _output(out, "raise NotImplementedError('User model was "
+                "not saved by save_all().'", indent=1)
+        _output_nl(out)
         return
 
-    msg = "Found user model '{}'; ".format(modelname) + \
+    msg = f"Found user model '{modelname}'; " + \
           "please check it is saved correctly."
     warning(msg)
 
     # Ensure the message is also seen if the script is run.
-    _output('print("{}")'.format(msg), fh)
+    _output(out, f'print("{msg}")')
 
-    _output(_reindent(pycode), fh)
-    cmd = 'load_user_model({}, "{}")'.format(
-        mod.calc.__name__, modelname)
-    _output(cmd, fh)
+    _output(out, _reindent(pycode))
```

Comment:
If this suggestion is applied it also needs a `import textwrap` at the top.

```suggestion
    textwarp.dedent(out, _reindent(pycode))
```

The behaviour is not exactly identical to `_reindent`, e.g.
```
    def f(a):
# Not sure why I would write this
        return a    
```
would be detended in line 1 and 3 by `_reindent`, but not by `textwrap.detent`. On the other hand, `_reindent` above (I'm making the comment here because this is the only line that uses it, so if we switch to `textwarp.detend` the entire definition of `_reindent` could be removed) also has a comment that it's fragile, just in other ways, for example if the first line does not start with "def":
```Python
    @some_decorator
    def func(a):
        pass
```
or
```
    class A():
        def __call__(self, a):
            return a
```
would be treated correctly by `textwrap.detend`, but not by `_reindent`. I tried one of the the example, and we could actually get such a case here:
```Python
In [3]: inspect.getsource(func)
Out[3]: '@functools.cache\ndef func(a):\n    return a\n'
```

On balance, I don't care. I'm always for removing code and using the standard library where possible, but I'm not going to argue that this problem is better than that problem.

Reply:
The problem with this code is that I haven't been able to easily trigger it, so if we can replace code with standard-specific stuff - I was looking when I wrote it originally but missed `textwrap`. As I always do.